### ISSUE TITLE
Tesla Model S not connecting via Bluetooth

### ISSUE THREAD

**Issue by patrickandri** [itsMeDavidV/Watchla-Support#95]
My Tesla Model S (2018) ist not connecting via Bluetooth. What should I do that this great feature is also working?

**Comment by itsMeDavidV** [comment]
Hi @patrickandri sorry to hear about this. Currently Watchla connecting via Bluetooth as a key is limited to only Tesla models that support using your Phone as key via Bluetooth.

Unfortunately it does not seem the 2018 Model S supports this. Are you able to use your phone as a key for your 2018 Tesla Model S?

**Comment by patrickandri** [comment]
Thanks for your prompt reply.
Yes, I can use my iPhone as a key for my Tesla Model S.

________________________________
Von: David Villegas ***@***.***>
Gesendet: Freitag, Januar 20, 2023 7:43 PM
An: itsMeDavidV/Watchla-Support ***@***.***>
Cc: patrickandri ***@***.***>; Mention ***@***.***>
Betreff: Re: [itsMeDavidV/Watchla-Support] Tesla Model S not connecting via Bluetooth (Issue #95)


Hi @patrickandri<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpatrickandri&data=05%7C01%7C%7C53d15bf259664843546408dafb16298b%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638098369842924607%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=YCtv6QlU6GRZyWM%2BIB2gZ%2FX%2BTVZdaOGUyNzIr%2F%2Br5r0%3D&reserved=0> sorry to hear about this. Currently Watchla connecting via Bluetooth as a key is limited to only Tesla models that support using your Phone as key via Bluetooth.

Unfortunately it does not seem the 2018 Model S supports this. Are you able to use your phone as a key for your 2018 Tesla Model S?

**Comment by itsMeDavidV** [comment]
Hi @patrickandri you're very welcome. So unfortunately after further research, it does appear that the 2018 Tesla Model S does not support Phone Key (which is using your Phone as a bluetooth key, different from controlling your vehicle via your phone over internet). 

You can confirm this in your car by navigating to your vehicle's settings and then going to "Locks". If you do not see your phone in your "Locks" list chances are your vehicle doesn't support Phone Key. Please do let me know if you're able to add your phone to the Locks list as that would indicate differently.

Here's a Tesla link that better defines a Phone Key and explains how to use and set one up: https://www.tesla.com/ownersmanual/models/en_us/GUID-E004FAB7-1C71-448F-9492-CACF301304D2.html

If your Model S does not have the Phone Key set up process that's described in the above Tesla link then unfortunately it most likely does not support Phone Key.

If this is the case for you, I'm really sorry. And if you'd like I can give you an App Store promo code to access Watchla for free, so that you can secure a refund given that this compatibility criteria was not disclosed prior. (Watchla still has many other great features that are compatible with your vehicle, that I'd love for you to be able to enjoy) 

Watchla has updated it's FAQs to reflect this compatibility criteria and the app and App Store page for Watchla will be updated shortly to reflect this compatibility criteria as well.

**Comment by ttwajsb** [comment]
David, on my Model 3 that does support Phone key I'm able to see the brief BLE indicator, and to operate the locks, but it's not sending the Remote Start code properly.

**Comment by itsMeDavidV** [comment]
@ttwajsb when connected via Bluetooth, to start driving all you need to do is open the Watchla app. From there you'll be good to go!

**Comment by ttwajsb** [comment]
First try: https://share.icloud.com/photos/0d98ADLGdYdRAmlyOAfxe73cQ

Second try: https://share.icloud.com/photos/057a-UHSmnskw8Mn3D5XhOSKA

Lock Screen: https://share.icloud.com/photos/074iSlqRvuBiX26oewU9DsA4w

**Comment by ttwajsb** [comment]
Nope, still getting PIN pad whether the iPhone app or the Watch app are logged in and  active
Do you mean to say I should no longer need to spin the crown?

**Comment by ttwajsb** [comment]
Seemed to work just now, but without an acknowledgment on the Tesla screen that's been customary previously. Will try it out more and get back to you.

**Comment by itsMeDavidV** [comment]
@ttwajsb ah I see what you mean now. You are wanting RS to bypass Pin-to-Drive, correct? To my knowledge I don't believe that a Bluetooth key (like Watchla) alone can achieve this as Pin to Drive is meant to be an additional security step. This could be different for a specific BLE RS command. I'll investigate and get back to you on that.

Also did the RS command you sent go through? In the video it appears to have not confirmed? If confirmed, it should have initiated RS.

**Comment by ttwajsb** [comment]
Went to the store and then didn't work. Confirmed key showed as Active on Locks screen, but PIN pad required multiple times, even after Watch chimed. Same symptoms as videos sent earlier today.

**Comment by ttwajsb** [comment]
The whole point of using Watchla to start the car was to bypass the PIN, and that's what it was doing very happily in the brief weeks between discovering that VPN was the culprit in my earlier complaints, and the switch to BLE on the Watch. I'll try reverting back from Bluetooth and see if I can again enjoy the delight of hopping in my car and driving away without the bother of the PIN—because Watchla saved me that step.

**Comment by patrickandri** [comment]
Hi David,

thank you very much for your investigation and research. Please apologize my delayed reply. I'm from Switzerland and therefore in a different time zone (CET).
I checked in my Tesla but there is no Phone Key setup available (see attached prtscn, no "Locks" button).
No need to refund. Your offer is appreciated but I'm happy with the Watchla with all the other features and the quick support you provide.

Another request: On the Apple Watch the measuring widget are all in miles. Would be great to have the opportunity to change to kilometer (Km) which is the standard measuring unit for Europe.

Thanks
Patrick

Von: David Villegas ***@***.***>
Gesendet: Freitag, 20. Januar 2023 21:24
An: itsMeDavidV/Watchla-Support ***@***.***>
Cc: patrickandri ***@***.***>; Mention ***@***.***>
Betreff: Re: [itsMeDavidV/Watchla-Support] Tesla Model S not connecting via Bluetooth (Issue #95)


Hi @patrickandri<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpatrickandri&data=05%7C01%7C%7C5d9aed5147154a2ee33e08dafb24513d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638098430653486903%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=4EdXzQw1I0sD43SVCbvl2%2FIWbJGCau5EuLbcBb3YKw0%3D&reserved=0> you're very welcome. So unfortunately after further research, it does appear that the 2018 Tesla Model S does not support Phone Key (which is using your Phone as a bluetooth key, different from controlling your vehicle via your phone over internet).

You can confirm this in your car by navigating to your vehicle's settings and then going to "Locks". If you do not see your phone in your "Locks" list chances are your vehicle doesn't support Phone Key. Please do let me know if you're able to add your phone to the Locks list as that would indicate differently.

Here's a Tesla link that better defines a Phone Key and explains how to use and set one up: https://www.tesla.com/ownersmanual/models/en_us/GUID-E004FAB7-1C71-448F-9492-CACF301304D2.html<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.tesla.com%2Fownersmanual%2Fmodels%2Fen_us%2FGUID-E004FAB7-1C71-448F-9492-CACF301304D2.html&data=05%7C01%7C%7C5d9aed5147154a2ee33e08dafb24513d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638098430653486903%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=ZZ7Sd3isWbGdx7Q8xP4AJQ74gRQLGI70M%2FyJ4FATWAI%3D&reserved=0>

If your Model S does not have the Phone Key set up process that's described in the above Tesla link then unfortunately it most likely does not support Phone Key.

If this is the case for you, I'm really sorry. And if you'd like I can give you an App Store promo code to access Watchla for free, so that you can secure a refund given that this compatibility criteria was not disclosed prior. (Watchla still has many other great features that are compatible with your vehicle, that I'd love for you to be able to enjoy)

Watchla has updated it's FAQs to reflect this compatibility criteria and the app and App Store page for Watchla will be updated shortly to reflect this compatibility criteria as well.

-
